### DISCUSSION TITLE
Train the unet from scratch

### DISCUSSION THREAD

**Discussion by dahui-y** [huggingface/diffusers#8458] [Q&A] [answered]
We know that we always fine-tune the unet based on the pre-trained model weights, which can be achieved by using method 'from_pretrained()' to load the weights. However, if we would like to train the model from scratch based on the diffuser repo, how do we implement it? Just input the config file and not use the method 'from_pretrained'?

The implemented code for fine-tuning is,
unet = UNet2DConditionModel.from_pretrained(
        args.pretrained_model_name_or_path, subfolder="unet", revision=args.non_ema_revision
    )

@sayakpaul

**Comment by sayakpaul** [comment, marked as answer]
You could do something like:

```python
from diffusers import UNet2DConditionModel

config = UNet2DConditionModel.load_config(
    "runwayml/stable-diffusion-v1-5", subfolder="unet"
)
unet = UNet2DConditionModel.from_config(config)
assert unet is not None
```
